### ISSUE TITLE
Image 8.0.37-debian(4fa43e731b20) Broken - World-writable config file '/etc/mysql/my.cnf' is ignored.

### ISSUE THREAD

**Issue by kitu-sbruce** [docker-library/mysql#1069]
```
➜  ~ docker run -it -e MYSQL_ROOT_PASSWORD=password--rm mysql:8.0.37-debian
2024-06-19 17:04:51+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.37-1debian12 started.
2024-06-19 17:04:51+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
2024-06-19 17:04:51+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 8.0.37-1debian12 started.
2024-06-19 17:04:51+00:00 [Note] [Entrypoint]: Initializing database files
mysqld: [Warning] World-writable config file '/etc/mysql/my.cnf' is ignored.
2024-06-19T17:04:51.569314Z 0 [System] [MY-013169] [Server] /usr/sbin/mysqld (mysqld 8.0.37) initializing of server in progress as process 82
2024-06-19T17:04:51.577881Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2024-06-19T17:04:52.279469Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
```

Does not occur with 8.0.36-debian.

**Comment by tianon** [comment]
Hmm, that's weird -- the permissions on that file shouldn't have changed :sob:

That file comes from:

https://github.com/docker-library/mysql/blob/1a703318803fa85bf69cb5d2e5b3f1c4087627b5/8.0/Dockerfile.debian#L94

Which is a line that hasn't changed in 7 years, so any change here is going to be something in how BuildKit/Docker are building it. :sob:

**Comment by tianon** [comment]
Definitely a change in behavior between BuildKit version 0.12 vs 0.13 -- I'm working on a reproducer so I can report it upstream. :sob: :disappointed:

**Comment by tianon** [comment]
Filed at https://github.com/moby/buildkit/issues/5066 now.

**Comment by aJanuary** [comment]
fwiw, this seems to be resolved in 8.0.38-debian

**Comment by tianon** [comment]
Ah nice, yes it is, thank you!

**Comment by marc-mabe** [comment]
@aJanuary 

> fwiw, this seems to be resolved in 8.0.38-debian

I can confirm it's fixed with `8.0.38-debian`.

Any chance to get the version `8.0.37-debian` fixed as well ?
-> We are using this for local development and are on RDS else which does not have 8.0.38 until now.

**Comment by tianon** [comment]
Unfortunately no, we typically do not rebuild older (unsupported) versions/tags.
